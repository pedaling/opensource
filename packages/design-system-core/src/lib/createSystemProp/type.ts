import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
