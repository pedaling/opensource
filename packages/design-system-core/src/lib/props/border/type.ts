import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
