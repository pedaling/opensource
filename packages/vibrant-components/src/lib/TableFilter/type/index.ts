export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
