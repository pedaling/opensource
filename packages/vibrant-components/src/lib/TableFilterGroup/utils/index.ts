import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
