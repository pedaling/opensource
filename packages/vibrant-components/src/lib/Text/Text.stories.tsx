import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Text } from './Text';

export default {
  title: 'Text',
  component: Text,
  args: {
    children: 'Sample Text',
    kind: 'title1',
  },
} as ComponentMeta<typeof Text>;

export const Basic: ComponentStory<typeof Text> = props => <Text {...props} />;
