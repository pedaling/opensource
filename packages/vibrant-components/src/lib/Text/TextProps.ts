import type { ReactNode } from 'react';
import type { ResponsiveValue, TypographyProps } from '@vibrant-ui/core';
import { withVariation } from '@vibrant-ui/core';
import type { TypographyKind, TypographyWeight } from '@vibrant-ui/theme';

export type TextProps = Omit<TypographyProps, 'typography' | 'fontWeight'> & {
  as?: 'span' | 'p';
  kind: ResponsiveValue<TypographyKind>;
  weight?: ResponsiveValue<TypographyWeight>;
  children: ReactNode;
};

export const withTextVariation = withVariation<TextProps>()();
