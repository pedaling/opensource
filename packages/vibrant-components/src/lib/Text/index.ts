export { Text } from './Text';
export { TextProps } from './TextProps';
