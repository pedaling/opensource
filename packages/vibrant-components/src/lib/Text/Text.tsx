import { Box } from '@vibrant-ui/core';
import { withTextVariation } from './TextProps';

const newlineRegex = /(\r\n|\n\r|\r|\n|\\n)/g;

const nl2br = (value: any) => {
  if (typeof value !== 'string') {
    return value;
  }

  return value.split(newlineRegex).map((line, index) => {
    if (line.match(newlineRegex)) {
      return <Box as="br" key={index} />;
    }

    return line;
  });
};

export const Text = withTextVariation(({ innerRef, as = 'span', kind, weight, children, ...restProps }) => (
  <Box as={as} ref={innerRef} typography={kind} fontWeight={weight} {...restProps}>
    {nl2br(children)}
  </Box>
));
