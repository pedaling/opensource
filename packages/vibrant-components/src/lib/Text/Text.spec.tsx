import type { ReactRenderer } from '@vibrant-ui/utils/testing';
import { createReactRenderer } from '@vibrant-ui/utils/testing';
import { Text } from './Text';

describe('<Text />', () => {
  const { render } = createReactRenderer();
  let renderer: ReactRenderer;

  describe('when title1 Text rendered', () => {
    beforeEach(() => {
      renderer = render(
        <Text kind="title1" weight="bold">
          Something
        </Text>
      );
    });

    it('span element rendered', () => {
      expect(renderer.container.querySelector('span')).toBeTruthy();
    });

    it('element have text', () => {
      expect(renderer.container.querySelector('span')?.textContent).toBe('Something');
    });

    it('element have title1 style', () => {
      const element = renderer.container.querySelector('span');

      expect(element).toHaveStyleRule('font-size', '2.125rem');

      expect(element).toHaveStyleRule('font-weight', '700');

      expect(element).toHaveStyleRule('line-height', '2.75rem');
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when Text with newline children rendered', () => {
    beforeEach(() => {
      renderer = render(<Text kind="title1">{'Some\nThing'}</Text>);
    });

    it('br tag created', () => {
      expect(renderer.container.querySelector('br')).toBeTruthy();
    });

    it('element have text', () => {
      expect(renderer.container.querySelector('span')?.textContent).toBe('SomeThing');
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
