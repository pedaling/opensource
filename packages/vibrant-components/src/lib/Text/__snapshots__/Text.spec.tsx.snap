// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> when Text with newline children rendered match snapshot 1`] = `
.emotion-1 {
  font-family: Pretendard;
  font-size: 2.125rem;
  font-weight: 700;
  line-height: 2.75rem;
}

<div>
  <span
    class="emotion-1"
  >
    Some
    <br
      class="emotion-1"
    />
    Thing
  </span>
</div>
`;

exports[`<Text /> when title1 Text rendered match snapshot 1`] = `
.emotion-0 {
  font-family: Pretendard;
  font-size: 2.125rem;
  font-weight: 700;
  line-height: 2.75rem;
}

<div>
  <span
    class="emotion-0"
  >
    Something
  </span>
</div>
`;
